sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -370,13 +372,11 @@ class ParquetFileFormat
       } else {
         logDebug(s"Falling back to parquet-mr")
         // ParquetRecordReader returns UnsafeRow
-        val reader = pushed match {
-          case Some(filter) =>
-            new ParquetRecordReader[UnsafeRow](
-              new ParquetReadSupport,
-              FilterCompat.get(filter, null))
-          case _ =>
-            new ParquetRecordReader[UnsafeRow](new ParquetReadSupport)
+        val reader = if (pushed.isDefined && enableRecordFilter) {
```

Comment:
Will this also disables parquet native row group filtering?

Reply:
Nope, I am sure it still enables group filtering. I will double check this and be back within few days.

Reply:
@jiangxb1987 If you do not like the solution here, could you submit the one you propose?

Reply:
Hm, I am active here. Could you share what's the problem here in this solution you'd imagine and discuss first?

I think there's no point of disabling row group filtering and, @jiangxb1987 asked if it actually disables row group filtering too, which might downgrade the performance. Current change does not do this.

I added a test for this concern - https://github.com/apache/spark/pull/15049#discussion_r147573054.

Reply:
If we want to disable both, I think we can simply disable Parquet predicate pushdown BTW.

Reply:
It should be fine to make this change, I was thinking we could make this change by setting the value of `ParquetInputFormat.RECORD_FILTERING_ENABLED` to false. Both way works and I don't have strong preference. Sorry for the late response.

Reply:
Oh, I have been looking at the JIRA as well but I thought that only exists in 1.9.0 as specified in the JIRA, also given https://github.com/apache/spark/pull/14671#issuecomment-240472754. Looks that flag also exists in Parquet 1.8.2 as well.

Yup, I don't have strong preference too.